ggshield/core/config/user_config.py

```diff
@@ -27,8 +28,27 @@
 CURRENT_CONFIG_VERSION = 2
 
 
-@dataclass
-class SecretConfig:
+@marshmallow_dataclass.dataclass
+class BaseConfig:
+    @classmethod
+    @pre_load(pass_many=False)
+    def filter_fields(cls, data: Dict, **kwargs: Any) -> Dict:
+        """
+        Remove and alert on unknown field.
+        """
+        field_names = {field_.name for field_ in fields(cls)}
+        filtered_fields = {}
+        for key, item in data.items():
+            if key in field_names:
+                filtered_fields[key] = item
+            else:
+                click.echo("Unrecognized key in config: {}".format(key))
```

Comment:
Can you use `display_warning()` here instead of `click.echo()`? `display_warning()` applies coloring and does not break pipe redirection because it prints to stderr instead of stdout.